### ISSUE TITLE
Installation Failure: Invalid Version Specifier in `setup.py` for `extract-msg<=0.29.*`

### ISSUE THREAD

**Issue by shrijayan** [deanmalmgren/textract#533]
### Describe the bug
When trying to install textract==1.6.4, the installation fails due to an issue with the install_requires field in the setup.py file. Specifically, the extract-msg<=0.29.* entry is not a valid version specifier because the .* suffix can only be used with the == or != operators.

### To Reproduce
1. Run pip install textract==1.6.4 in a Python environment.
2. Observe the error during the installation process when building the wheel.

### Expected behavior
The package should install successfully without errors.

### Error Output
error in textract setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement 

```sh
specifiers; .* suffix can only be used with `==` or `!=` operators
extract-msg<=0.29.*
             ~~~~~~~^
```

Desktop (please complete the following information):
- OS: MacOS
- Textract version: 1.6.4
- Python version: 3.11
- Virtual environment: yes

### Additional context
It seems like the issue is related to invalid version specifiers in the setup.py file of the textract package. A potential fix would be to update the version specifier to extract-msg<=0.29 or resolve the versioning conflict.

**Comment by jkfran** [comment]
I am having this issue as well.

**Comment by GHUFRAN-HYDER** [comment]
**Solution for Installation Failure Due to Invalid Version Specifier in `setup.py` for `extract-msg<=0.29.*`**

Hi @shrijayan @jkfran,
The issue you're encountering is caused by an invalid version specifier in the requirements/python file being parsed by setup.py. The version constraint `extract-msg<=0.29.*` is invalid because Python packaging tools like `pip` only support `==` or `!=` with the `.*` suffix.

### Proposed Fix:
To fix this issue, modify the requirements/python file as follows:

1. Find the line that contains:
   ```
   extract-msg<=0.29.*
   ```
2. Modify it to:
   ```
   extract-msg<=0.29
   ```
   This correction ensures that the version specifier is in a valid format.

3. After modifying, you can proceed with the installation using the following command:
   ```bash
   python setup.py install
   ```


This should resolve the issue and allow you to proceed with the installation without errors. Let me know if this helps or if further assistance is needed!

Best regards,

**Comment by NGeorgescu** [comment]
It should be <0.29.6

for now I put the line in my requirements.txt my own fork:

    git+https://github.com/ngeorgescu/textract@403943b#egg=textract

not sure when this will be fixed given that there are multiple PRs for this one-character fix.

**Comment by zerolab** [comment]
I believe this was fixed in #543 and out with the v2 release.
https://github.com/deanmalmgren/textract/blob/master/pyproject.toml#L22

cc @KyleKing

**Comment by KyleKing** [comment]
Yes! The dependency issues should be resolved. Let me know if you run into any issues now that I'm involved as a maintainer!
